common/http/class.Request.php

```diff
@@ -244,8 +249,20 @@ public function send()
         $httpResponse->headerOut = curl_getinfo($curlHandler, CURLINFO_HEADER_OUT);
         $httpResponse->effectiveUrl = curl_getinfo($curlHandler, CURLINFO_EFFECTIVE_URL);
         $httpResponse->responseData = $responseData;
+
+        $fullInfo = curl_getinfo($curlHandler);
+
         //curl_setopt($curlHandler, );
         curl_close($curlHandler);
+
+        if ($followRedirects && in_array($httpResponse->httpCode, self::REDIRECT_CODES, true)) {
```

Comment:
Don't you also need to set `CURLOPT_FOLLOWLOCATION` flag to 0 in order to disable curl-level redirects?

Reply:
Copying also there the discussion in PM:
It shouldn't be set directly to `false`/`0` as it is either undeclared or already set to `false` in curl source.
However, I do fully agree with  the raised point, and to make sure the behaviour won't change - I'm adding this option too.
Thank you!